web/actions/form_submissions_retriever.ex

```diff
@@ -0,0 +1,22 @@
+defmodule Formerer.FormSubmissionsRetriever do
+  import Ecto.Query, only: [where: 2, where: 3, order_by: 3, first: 1]
+  alias Formerer.{Submission, Repo}
+
+  def get_form_submissions(form, last_seen_id) do
+    last_seen = last_seen_submission_date(last_seen_id)
+    Submission
+    |> where(form_id: ^form.id)
+    |> where([submission], submission.inserted_at > ^last_seen)
+    |> order_by([s], [desc: s.inserted_at])
+    |> Repo.all
+  end
+
+  defp last_seen_submission_date(nil) do
+    submission = Submission |> first |> Repo.one
```

Comment:
For both of these where we are only after a single value we can select the single value from the database instead of filling up the whole struct:

For instance:

`Submission |> select([s], s.inserted_at) |> first |> Repo.one`

This will only return the inserted_at value